packages/clerk-js/src/core/resources/Session.ts

```diff
@@ -84,7 +85,10 @@ export class Session extends BaseResource implements SessionResource {
     const path = template ? `${this.path()}/tokens/${template}` : `${this.path()}/tokens`;
     const tokenResolver = Token.create(path);
     SessionTokenCache.set({ tokenId, tokenResolver });
-    return tokenResolver.then(res => res.getRawString());
+    return tokenResolver.then(token => {
+      eventBus.dispatchEvent(new CustomEvent('token:update', { detail: { token } }));
```

Comment:
🔧 I'd suggest abstracting the CustomEvent via a createEvent helper that belongs to the eventBus module. That way, we can change the internal implementation of the event bus without touching all the event emission points.

Reply:
Since the `addEventListener` will depend on the `CustomEvent` to parse the data from `event` argument should i also introduce a `eventBus.parseEvent` ?

Reply:
Yes. Or just introduce an `on` function.